common/remote/sockfile.cpp

```diff
@@ -5745,13 +5745,29 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
             }
         }
 
-        if (_useSSL)
+        if (_connectMethod == SSLOnly || _connectMethod == SSLFirst || _connectMethod == UnsecureFirst)
         {
             if (sslep.port == 0)
-                throw createDafsException(DAFSERR_connection_failed, "Secure dafilesrv port not specified");
+                throw createDafsException(DAFSERR_serverinit_failed, "Secure dafilesrv port not specified");
 
-            if ( (_useSSL != UnsecureFirst) && (!securitySettings.certificate || !securitySettings.privateKey) )
-                throw createDafsException(DAFSERR_connection_failed, "SSL Certificate and/or Key file information not found in environment.conf");
+            if (_connectMethod == UnsecureFirst)
```

Comment:
It will still listen on the SSL port but it will close a connection on that port quickly so that client can failover to non-ssl faster than a timeout - if configured to do so.  This is different than SSLNone where it does not listen on the SSL port.
I thought it was finer control over things this way as without this code UnsecureFirst and SSLFirst are identical as far as dafilesrv is concerned.
But I agree it could cause some confusion.
I could remove this check or I could add log msgs at startup that dafilesrv is configured for both but is missing certificate info.